com.vagueentertainment.Framed.yml

```diff
@@ -0,0 +1,75 @@
+app-id: com.vagueentertainment.Framed
+runtime: org.freedesktop.Platform
+runtime-version: '20.08'
+sdk: org.freedesktop.Sdk
+command: framed
+modules: 
+  
+  - name: framed
+    buildsystem: simple
+    build-commands:
+      - install -D -m755 Framed.x86_64 /app/bin/Framed.x86_64
+      - install -D Framed.pck /app/bin/Framed.pck
+      - cp -r HoneyComb-redistributable /app/bin/HoneyComb-redistributable
+      - install -D -m755 framed.sh /app/bin/framed
+      - desktop-file-edit --remove-key=PrefersNonDefaultGPU $FLATPAK_ID.desktop
+      - desktop-file-edit --set-icon=$FLATPAK_ID $FLATPAK_ID.desktop
+      - install -Dm644 $FLATPAK_ID.desktop /app/share/applications/$FLATPAK_ID.desktop
+      - install -Dm644 icon.svg /app/share/icons/hicolor/scalable/apps/$FLATPAK_ID.svg
+      - install -Dm644 $FLATPAK_ID.appdata.xml /app/share/appdata/$FLATPAK_ID.appdata.xml
+            
+    sources:
+      - type: archive
+        url: https://github.com/VagueEntertainment/Framed/archive/refs/tags/v0.3.tar.gz
+        sha256: "1558912caf5d83347be02a598b59fa13e09ca758c1a35d18a4d407784292e667"
+        
+      - type: archive
+        url: https://github.com/VagueEntertainment/HoneyComb-redistributable/archive/refs/tags/0.1.1.tar.gz
+        sha256: "69177aaa166089ed14a41c1fe673abfcd08d0f57d2bdb62f524043d1e04ed5dc"
+        dest: HoneyComb-redistributable
+      
+      - type: script
+        dest-filename: framed.sh
+        commands:
+          - export APPDATA="$XDG_DATA_HOME"
+          - export XDG_DATA_HOME="$APPDATA"
+          - export HOME="$APPDATA"
+          - export PATH=$PATH:/app/bin/HoneyComb-redistributable/service
+          - /app/bin/Framed.x86_64 --path=/app/bin/
+          
+      - type: file
+        path: icon.svg
+        
+      - type: file
+        path: com.vagueentertainment.Framed.appdata.xml
+        
+      - type: file
+        path: com.vagueentertainment.Framed.desktop
+        
+  - modules/python3-pip.json      
+  - modules/python3-wheel.json
+  - modules/python3-semantic_version.json
+  - modules/python3-setuptools.json
+  - modules/python3-setuptools_rust.json
+  - modules/python3-setuptools_scm.json
+  - modules/python3-bottle.json
+  - modules/python3-requests.json
+  - modules/python3-netifaces.json
+  - modules/python3-cryptography.json
+  - modules/python3-pycrypto.json
+  - modules/python3-pyopenssl.json
+  - modules/python3-wcwidth.json
+  - modules/python3-prettytable.json
+  - modules/python3-steem.json
+  - modules/python3-hivepy.json
+  - modules/python3-websockets.json
+  - modules/python3-asyncio.json
+        
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --share=network
+  - --socket=pulseaudio
+  - --filesystem=home:ro
```

Comment:
Since you set `$HOME` in the launch script to `$XDG_DATA_HOME`, do you really need to give access to the whole home directory?

Reply:
Yes, the project can be set to display images in any directory. However If setting XDG_DATA_HOME is enough I can remove it. Will test and see what works best.

Reply:
You shouldn't have to set XDG_DATA_HOME. And you set HOME to something that's not HOME. So I'm not sure how this falls in place.

Reply:
Looks like both are needed at this stage in development.

Reply:
Ah you ninja'd me. I meant HOME not XDG_DATA_HOME in that first comment. I'm doing to many things at once it seems.

To explain things a bit further: 
I set both XDG_DATA_HOME and HOME to make sure that all data created by the app is put in the directories created by flatpak in the .var/app/<app name> directory structure. Without setting both Godot and possibly Python write to the users home directory as I would prefer all data created by the application be stored in the directory mentioned adding both is necessary at this time. 

The reason I then set --filesystem=home:ro is to give the users access to their real home directory in order to select a folder use as part of the gallery. Setting home to read only maintains some level of security and is a good way to make sure all data created by my app is indeed being created in its own directory structure.

Reply:
I think what you want is `--persist=` for the directories the app writes in the home directory, and leave $HOME alone.

Reply:
Yeah, ideally it should use `--persist` to "redirect" specific directories but I'd say it's not a blocker.